### PR TITLE
Standalone fault_fix object

### PR DESCRIPTION
**Pull request by irwiss**
#### Summary
None

#### Purpose of change

Work towards https://github.com/CleverRaven/Cataclysm-DDA/issues/64528

#### Describe the solution

https://github.com/CleverRaven/Cataclysm-DDA/pull/65197/commits/72a8af6bd0aee04c1733932e6da7098cb39283ba Makes fault_fix (previously mending_method) a standalone json object with it's own loader. Also slightly changes schema: skills are now a dictionary, this should allow cleaner copy-from if needed and less complex json.

https://github.com/CleverRaven/Cataclysm-DDA/pull/65197/commits/dd5cbc9071633ace07ba902539e269eb0473df01 Merges the existing fault_fix duplicates for dirt and blackpowder cleaning / lubrication now that fixes are separate from faults.

https://github.com/CleverRaven/Cataclysm-DDA/pull/65197/commits/4f17b9513e5c1fafbfd8910f97b5bfa4804d2c42 Moves hardcoded `(faulty)` prefixes into json definition of the fault

#### Testing

Most user facing changes should be string changes; `(faulty)` is now defined per-fault, so gun with a fault_gun_chamber_spent will display `jammed` instead of faulty, salvaged bionics will display `salvaged`, blackpowder fouled guns will display `rusting` etc

The layout of fault mending changed slightly as fault_fix now can add/remove multiple faults, unused so invisible yet but also a UI change: there's label for fault_fix modifying damage

Spawn lightly damaged cars until you get some engine faults, should be able to mend them if the requirements are fulfilled

Spawn 2 5.56 guns and some blackpowder ammo, load one with BP ammo, shoot both, see that the correct faults were added (one should have bp fouling, other normal fouling), mend both guns - see both guns lose their correspponding fouling fault

### CODE REVIEW

**Comment by ama0** [comment]
Maybe it's better to change the json filenames from `faults_xyz.json` to `xyz_faults.json` in `data/json/faults/`. Just to make it easier to navigate all the files, because there probably will be a lot of files in this folder. And I think it's more intuitive 😅

| faults_xyz  | xyz_faults |
| ------------- | ------------- |
| faults_bar | bar_faults  |
| faults_baz  | bar_fixes |
| faults_bionics  |  baz_faults |
| faults_foo  |  baz_fixes |
| faults_guns |  bionics_faults |
| faults_vehicles  |  bionics_fixes |
| fixes_bar | foo_faults |
| fixes_baz  | foo_fixes |
| fixes_bionics | guns_faults |
| fixes_foo | guns_fixes |
| fixes_guns | vehicle_faults |
| fixes_vehicles | vehicle_fixes |

**Comment by Fris0uman** [comment]
Don't this need some documentation to explain how to write a fault object?

**Comment by irwiss** [comment]
Yep forgot about those, seems docs didn't exist even for previous fault definitions either, added some

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -1344,6 +1344,46 @@ the item is spawned the variables set on the prototype no longer affect the item
...
+Faults can be defined for more specialized damage of an item.
+
+```C++
+{
+  "type": "fault",
+  "id": "fault_gun_chamber_spent", // unique id for the fault
+  "name": { "str": "Spent casing in chamber" }, // fault name for display
+  "description": "This gun currently...", // fault description
+  "item_prefix": "jammed", // optional string, items with this fault will be prefixed with this
+  "flags": [ "JAMMED_GUN" ] // optional flags, used by C++ parts
```
Comment (Fris0uman) [inline doc/JSON_INFO.md:1358]: Maybe add a link to the fault flags doc around here https://github.com/CleverRaven/Cataclysm-DDA/blob/a5245cacc5f4736bf272fce944dd9cf357483fe0/doc/JSON_FLAGS.md?plain=1#L1639

Comment (irwiss) [reply doc/JSON_INFO.md]: Yep, linked and removed outdated parts from there as well

**Pull request merged**